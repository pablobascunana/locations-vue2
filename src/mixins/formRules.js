import Vue from 'vue'

Vue.mixin({
  data() {
    return {
      formRules: {
        loginCredentials: true,
        required: value => !!value || this.$t('validations.required'),
        maxLength: value => value.length < 101 || this.$t('validations.maxLength'),
        validEmail: value => /(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@(([[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(value) || this.$t('validations.validEmail'),
        samePassword: value => (value == this.user.password) || this.$t('validations.samePassword'),
        sameRepeteadPassword: value => (value == this.user.repeteadPassword) || this.$t('validations.samePassword'),
      }
    }
  }
})