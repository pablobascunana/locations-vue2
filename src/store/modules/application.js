function initialApplicationState() {
  return {
    isLogin: true,
    accessToken: undefined,
    refreshToken: undefined,
    addLocationDialogVisible: false,
    removeLocationDialogVisible: false
  }
}

const application = {
  namespaced: true,
  state: initialApplicationState,
  mutations: {
    resetApplicationStore(state) {
      const initState = initialApplicationState();
      Object.keys(initState).forEach(key => {
        state[key] = initState[key];
      })
    },
    isLogin(state, visible) {
      state.isLogin = visible;
    },
    setAccessToken(state, token) {
      state.accessToken = `Bearer ${token}`;
    },
    setRefreshToken(state, token) {
      state.refreshToken = token;
    },
    isAddLocationDialogVisible(state, visible) {
      state.addLocationDialogVisible = visible;
    },
    isRemoveLocationDialogVisible(state, visible) {
      state.removeLocationDialogVisible = visible
    }
  }
};

export default application;
  