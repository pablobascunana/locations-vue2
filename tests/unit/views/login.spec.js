import { createLocalVue, shallowMount } from '@vue/test-utils'

import VueRouter from 'vue-router'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

import application from '@/store/modules/application.js'
import user from '@/store/modules/user.js'

import '@/mixins/formRules'
import Login from '@/views/Login.vue'

import * as loginMock from '../mocks/login.mock.js'

const Plugins = {
    install(Vue) {
      Vue.prototype.EventBus = new Vue();
    }
};

const localVue = createLocalVue();
localVue.use(Plugins);
localVue.use(VueRouter);
localVue.use(Vuex);
const router = new VueRouter();
let vuetify = new Vuetify();


describe('Login', () => {
    let wrapper;
    let store;
    let state = {};
    let actions = {};
    let modules = {
        application,
        user
    };
    const mocks = {
        EventBus: {
            $on: jest.fn(),
            $off: jest.fn(),
            $emit: jest.fn(),
        },
        $t: () => {'some specific text in any language'}
    };
    beforeEach(() => {
        store = new Vuex.Store({ state, actions, modules });

        wrapper = shallowMount(Login, {
            localVue,
            vuetify,
            router,
            store,
            data() {
                return {
                    user: {
                        userName: "test"
                    }
                }
            },
            mocks
        });
    });

    it("Should match snapshot", () => {
        expect(wrapper).toMatchSnapshot()    
    });

    it('Renders a vue instance', () => {
        expect(wrapper.isVueInstance()).toBe(true);
    });

    it('Click in register button', async () => {
        const button = wrapper.find('#register');
        await button.trigger('click');
        expect(button.attributes().disabled).toBe(undefined);
    });

    it('Click in login button', async () => {
        const button = wrapper.find('#login');
        await button.trigger('click');
        expect(button.attributes().disabled).toBe(undefined);
    });

    it('Call to set store and go to historical locations function', () => {
        const setStoreAndGoToHistoricalLocationsFn = jest.spyOn(wrapper.vm, 'setStoreAndGoToHistoricalLocations');
        wrapper.vm.setStoreAndGoToHistoricalLocations(loginMock.LOGIN_RESPONSE);
        expect(setStoreAndGoToHistoricalLocationsFn).toBeCalled();
        expect(wrapper.vm.$store.state.application.accessToken).toBe(`${'Bearer '}${loginMock.LOGIN_RESPONSE.access_token}`);
        expect(wrapper.vm.$store.state.application.refreshToken).toBe(loginMock.LOGIN_RESPONSE.refresh_token);
        expect(wrapper.vm.$store.state.application.isLogin).toBe(false);
        expect(wrapper.vm.$store.state.user.userUuid).toBe(loginMock.USER_ID);
        expect(wrapper.vm.$store.state.user.userName).toBe(loginMock.USER_DATA.userName);
    });

    it('Call to check what error is function', () => {
        wrapper.vm.$refs.loginForm.validate = () => true;
        const checkWhatErrorIsFn = jest.spyOn(wrapper.vm, 'checkWhatErrorIs');
        wrapper.vm.checkWhatErrorIs(401);
        wrapper.vm.checkWhatErrorIs(404);
        wrapper.vm.checkWhatErrorIs(500);
        expect(checkWhatErrorIsFn).toBeCalled();
    });
});