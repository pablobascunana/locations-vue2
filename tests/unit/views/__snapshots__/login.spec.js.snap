// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should match snapshot 1`] = `
<v-form>
  <h1 class="text-center pb-12"></h1>
  <v-row class="justify-center">
    <v-col xl="4" lg="4" md="6" sm="6" cols="8">
      <v-text-field id="userName" prepend-icon="mdi-email-outline" rules="true" type="text"></v-text-field>
    </v-col>
  </v-row>
  <v-row class="justify-center">
    <v-col xl="4" lg="4" md="6" sm="6" cols="8">
      <v-text-field id="password" autocomplete="off" prepend-icon="mdi-lock-outline" rules="true" type="password"></v-text-field>
    </v-col>
  </v-row>
  <v-row class="justify-center mt-6">
    <v-btn id="login" color="primary" elevation="0">
    </v-btn>
  </v-row>
  <v-row class="justify-center mt-12">
    <v-btn id="register" text="">
    </v-btn>
  </v-row>
</v-form>
`;
